server/backend/check.ml

```diff
@@ -232,19 +243,35 @@ let get_obuilder ~conf ~opam_repo ~opam_repo_commit ~extra_repos switch =
   in
   let open Obuilder_spec in
   let cache = cache ~conf in
-  let from = match Server_configfile.platform_os conf with
-    | "linux" -> Server_configfile.platform_image conf
+  let os = Server_configfile.platform_os conf in
+  let distribution = Server_configfile.platform_distribution conf in
+  let from = match os with
+    | "linux" -> "ocaml/opam:"^distribution (* typically this is 'debian-unstable' which is 5.0.0 *)
+    | "freebsd" -> distribution^"-ocaml-5.0" (* Either 4.14.1 or 5.0.0 could be selected *)
+    | "macos" -> "macos-"^distribution^"-ocaml-5.0" (* Either 4.14.1 or 5.0.0 could be selected *)
     | os -> failwith ("OS '"^os^"' not supported") (* TODO: Should other platforms simply take the same ocurrent/opam: prefix? *)
   in
+  let ln_opam = match os with
+    | "linux" -> "sudo ln -f /usr/bin/opam-dev /usr/bin/opam"
+    | "freebsd" -> "sudo ln -f /usr/local/bin/opam-2.1 /usr/local/bin/opam"
+    | "macos" -> "ln -f ~/local/bin/opam-2.1 ~/local/bin/opam"
```

Comment:
Same here could we have `opam-2.2` on FreeBSD and macOS? We can change opam-dev to `opam-2.2` on linux if needed if the dev version can't be done

Reply:
I have updated the base images on macOS and FreeBSD to include `opam-dev` as well as `opam-2.1`.  I have updated the PR to set these to `opam-dev`.